wpimath/src/main/java/edu/wpi/first/math/trajectory/TrapezoidProfile.java

```diff
@@ -128,8 +128,8 @@ public State calculate(double t, State current, State goal) {
     m_current = direct(current);
     goal = direct(goal);
 
-    if (m_current.velocity > m_constraints.maxVelocity) {
-      m_current.velocity = m_constraints.maxVelocity;
+    if (Math.abs(m_current.velocity) > Math.abs(m_constraints.maxVelocity)) {
```

Comment:
A lot more of this algorithm assumes the max velocity constraint is positive. Instead of using `abs()` everywhere, it would be cleaner to throw in the constructor for negative constraints, then just assume they're positive.